model/waterfall.go

```diff
@@ -13,54 +13,111 @@ import (
 )
 
 const (
+	DefaultWaterfallQueryCount   = 20
 	DefaultWaterfallVersionLimit = 5
 	MaxWaterfallVersionLimit     = 300
 )
 
 type WaterfallTask struct {
 	Id          string `bson:"_id" json:"_id"`
-	DisplayName string `bson:"display_name" json:"display_name,omitempty"`
-	Status      string `bson:"status" json:"status,omitempty"`
+	DisplayName string `bson:"display_name" json:"display_name"`
+	Status      string `bson:"status" json:"status"`
 }
 
 type WaterfallBuild struct {
 	Id          string          `bson:"_id" json:"_id"`
-	Activated   bool            `bson:"activated" json:"activated,omitempty"`
-	DisplayName string          `bson:"display_name" json:"display_name,omitempty"`
-	Version     string          `bson:"version" json:"version,omitempty"`
-	Tasks       []WaterfallTask `bson:"tasks" json:"tasks,omitempty"`
+	Activated   bool            `bson:"activated" json:"activated"`
+	DisplayName string          `bson:"display_name" json:"display_name"`
+	Version     string          `bson:"version" json:"version"`
+	Tasks       []WaterfallTask `bson:"tasks" json:"tasks"`
 }
 
 type WaterfallBuildVariant struct {
 	Id          string           `bson:"_id" json:"_id"`
-	DisplayName string           `bson:"display_name" json:"display_name,omitempty"`
-	Builds      []WaterfallBuild `bson:"builds" json:"builds,omitempty"`
+	DisplayName string           `bson:"display_name" json:"display_name"`
+	Builds      []WaterfallBuild `bson:"builds" json:"builds"`
 }
 
 type WaterfallOptions struct {
 	Limit      int      `bson:"-" json:"-"`
+	MaxOrder   int      `bson:"-" json:"-"`
+	MinOrder   int      `bson:"-" json:"-"`
```

Comment:
Might help to clarify that these are exclusive bounds and that passing `MinOrder` reverses the sorting direction.

Reply:
`MinOrder` doesn't reverse the sorting, sorry if that was confusing. I added a comment, let me know if that clarifies.